components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml

```diff
@@ -4439,24 +4443,42 @@ document.observe("xwiki:dom:loaded", function() {
         this.monthSelected();
         this.updateDate();
     },
-    
+
     createYearDropdown : function() {
       this.yearSelector = new widgets.FuzzyDatePickerDropdown({name: "year"});
 
       var today = new Date();
       var crtYear = today.getYear() + 1900;
       var startYear = 1900;
 
+      var range = (this.__date.hasOwnProperty("range") &amp;&amp; this.__date.range.hasOwnProperty("years")) ? this.__date.range.years : 1;
+      var year = this.__date.hasOwnProperty("year") ? this.__date.year : null;
       var values = [];
       for (var y = crtYear; y &gt;= startYear; --y) {
-        values.push({"value" : y, "selected" : this.__date.year == y});
+        values.push({"value" : y, "selected" : ((year == y) &amp;&amp; (range &lt;= 1)) });
         if (y % 10 == 0) {
-            values.push({"value" : (y + "s"), "cssClass" : "decade", "text" : (y + 's'), "selected" : this.__date.decade == y + "s"});
+            values.push({ "value" : (y + "s"),
+                          "cssClass" : "decade",
+                          "text" : (y + 's'),
+                          "selected" : ((year == y) &amp;&amp; (range &gt; 1))});
         }
       }
-      values.push({"value": "1800s", "cssClass" : "decade", "selected" : this.__date.decade == "1800s"});
-      values.push({"value": "1700s", "cssClass" : "decade", "selected" : this.__date.decade == "1700s"});
-      values.push({"value": "1600s", "cssClass" : "decade", "selected" : this.__date.decade == "1600s"});
+
+      var insertCentury = function(centuryStart) {
+          var centuryEnd = centuryStart + 100;
+          if (year != null &amp;&amp; year &gt;= centuryStart &amp;&amp; year &lt; centuryEnd &amp;&amp; range == 1) {
+              values.push({"value" : year, "selected" : true });
+          }
+          var yearInCentury = (year != null) &amp;&amp; (year &gt;= centuryStart) &amp;&amp; (year &lt; centuryEnd) &amp;&amp; (range &gt; 1);
+          values.push({"value": centuryStart + "s", "cssClass" : "century", "selected" : yearInCentury});
+      }
+      insertCentury(1800);
+      insertCentury(1700);
+      insertCentury(1600);
+      insertCentury(1500);
+      if (year != null &amp;&amp; year &lt; 1500) {
```

Comment:
this "push" just adds a custom year, and if the year is less than 1500, it must be the last one in the list, so we push it last. If it is before that, it would be pushed before